### PR TITLE
ENG-12541 sync repair items in repair log between master and replicas

### PR DESCRIPTION
**Pull request by zeemanhe**
Transaction messages are queued in repair log before they have chance to get updated transaction id on partition leaders, which may cause two issues: (1)committed transactions may be re-played in the repair process; (2) The item stored in the repair log has starting transaction id of long.min, which alway points to partition 0.  In the repair process, partition mis-matches may occur.
This issue is addressed in the pull request.

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/InitiatorMailbox.java [hunk 1]
```diff
@@ -307,9 +307,12 @@ else if (message instanceof RepairLogTruncationMessage) {
             m_repairLog.deliver(message);
             return;
         }
-        m_repairLog.deliver(message);
+
         if (canDeliver) {
+            //Repair log will be handled with updated sp handle in Scheduler in this case
             m_scheduler.deliver(message);
+        } else {
+            m_repairLog.deliver(message);
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/iv2/InitiatorMailbox.java]: Can you add a comment to explain what does this mean?

The repairLog only cares about the message that can not be delivered?

Comment (zeemanhe) [inline src/frontend/org/voltdb/iv2/InitiatorMailbox.java]: For delivered messages, SpScheduler will log them to  repair log when updated txn ids are available.

**Comment by xinjiacs** [comment]
@zeemanhe Are you missing handling `DumpMessage` or `RepairLogTruncationMessage`?

File: src/frontend/org/voltdb/iv2/SpScheduler.java [hunk 2]
```diff
@@ -166,7 +166,7 @@ public String toString() {
     long m_lastSentTruncationHandle = Long.MIN_VALUE;
     // the max schedule transaction sphandle, multi-fragments mp txn counts one
     long m_maxScheduledTxnSpHandle = Long.MIN_VALUE;
-
+    protected RepairLog m_repairLog;
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: @zeemanhe Add a comment here for the `m_repairLog ` explaining where it gets initialized and what it is.

Comment (zeemanhe) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: sure

**Comment by zeemanhe** [comment]
"Are you missing handling DumpMessage or RepairLogTruncationMessage?"
DumpMessage does not do anything in RepairLog. RepairLogTruncationMessage is handled specifically in InitiatorMailbox.

**Pull request merged**